src/Core/RulesAccessTrait.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+/**
+ * @file
+ * Contains \Drupal\rules\Core\RulesAccessTrait.
+ */
+
+namespace Drupal\rules\Core;
+
+use Drupal\Core\Access\AccessResult;
+use Drupal\Core\Session\AccountInterface;
+
+/**
+ * Implements access related functions for Rules plugins.
+ */
+trait RulesAccessTrait {
+
+  /**
+   * Check administration access.
+   *
+   * @param AccountInterface $account
+   *   (optional) The user for which to check access, or NULL to check access
+   *   for the current user. Defaults to NULL.
+   * @param bool $return_as_object
+   *   (optional) Defaults to FALSE.
+   *
+   * @return bool|\Drupal\Core\Access\AccessResultInterface
+   *   The access result. Returns a boolean if $return_as_object is FALSE (this
+   *   is the default) and otherwise an AccessResultInterface object.
+   *   When a boolean is returned, the result of AccessInterface::isAllowed() is
+   *   returned, i.e. TRUE means access is explicitly allowed, FALSE means
+   *   access is either explicitly forbidden or "no opinion".
+   */
+  public function hasAdminAccess(AccountInterface $account = NULL, $return_as_object = FALSE) {
```

Comment:
regular d8 wording is "checkAccess", thus should be checkConfigurationAccess() (discussed with klausi)